packages/ui/src/app/pages/Settings/SettingsNotificationsTab.tsx

```diff
@@ -0,0 +1,192 @@
+import { EmailSubscriptionModalCall } from '@/memberships/modals/EmailSubscriptionModal'
+import {
+  useGetBackendMeQuery,
+  useUpdateBackendMemberMutation,
+} from '@/memberships/queries/__generated__/backend.generated'
+
+import { SettingsLayout } from './SettingsLayout'
+
+type SettingsNotificationsTabFormData = {
+  email: string
+  receiveEmailNotifications: boolean
+}
+
+const SettingsNotificationsTabFormSchema = Yup.object().shape({
+  email: Yup.string().email().required('This field is required.'),
+  receiveEmailNotifications: Yup.boolean(),
+})
+
+export const SettingsNotificationsTab: FC = () => {
+  const {
+    activeMemberExistBackendData,
+    activeMemberExistBackendError,
+    activeMemberExistBackendLoading,
+    activeMemberExistBackendRefetch,
+    backendClient,
+    activeMemberSettings,
+  } = useNotificationSettings()
+  const { showModal } = useModal()
+
+  const form = useForm<SettingsNotificationsTabFormData>({
+    defaultValues: {
+      email: '',
+      receiveEmailNotifications: true,
+    },
+    resolver: useYupValidationResolver<SettingsNotificationsTabFormData>(SettingsNotificationsTabFormSchema),
+  })
+  const {
+    handleSubmit,
+    reset,
+    formState: { isDirty, dirtyFields },
+    watch,
+  } = form
+
+  const { data: meData, refetch: meRefetch } = useGetBackendMeQuery({
+    client: backendClient,
+    onCompleted: (data) => {
+      reset({
+        email: data.me?.unverifiedEmail ?? data.me?.email ?? '',
+        receiveEmailNotifications: data.me?.receiveEmails ?? true,
+      })
+    },
+    skip: !activeMemberSettings?.accessToken,
+  })
+
+  const [sendUpdateMemberMutation] = useUpdateBackendMemberMutation({
+    client: backendClient,
+  })
+  const [newLinkGenerated, setNewLinkGenerated] = useState(false)
+  const [showSettingsUpdatedModal, setShowSettingsUpdatedModal] = useState(false)
+
+  const handleSubscribeClick = () => {
+    // TODO: this will show member select if there is no active membership.
+    // However, if the user selects a membership that's already registered, the subscription modal will still show.
+    showModal<EmailSubscriptionModalCall>({
+      modal: 'EmailSubscriptionModal',
+      data: {
+        onSubscribe: () => {
+          meRefetch()
+          activeMemberExistBackendRefetch()
+        },
+      },
+    })
+  }
+
+  const handleGenerateNewLinkClick = () => {
+    sendUpdateMemberMutation({
+      variables: {
+        email: meData?.me?.unverifiedEmail ?? meData?.me?.email ?? '',
+      },
+    })
+    setNewLinkGenerated(true)
+  }
+
+  const handleSaveChangesClick = handleSubmit(async (data) => {
+    await sendUpdateMemberMutation({
+      variables: {
+        email: dirtyFields.email ? data.email : undefined,
+        receiveEmails: dirtyFields.receiveEmailNotifications ? data.receiveEmailNotifications : undefined,
+      },
+    })
+    setShowSettingsUpdatedModal(true)
+  })
+
+  const loadingContent = 'Loading notification settings'
```

Comment:
```suggestion
  const loadingContent = <Loading text="Loading notification settings" />
```

Reply:
I was planning to do that as part of issue with handling errors, but thanks for your suggestions, applied.